core/src/main/scala/chisel3/Module.scala

```diff
@@ -239,12 +275,11 @@ package internal {
       rec(start)
     }
 
-    private[chisel3] def cloneIORecord(proto: BaseModule)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): ClonePorts = {
+    private[chisel3] def createIORecord[T <: BaseModule](cloneParent: ModuleClone[T])(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): ClonePorts = {
```

Comment:
In this function we check if proto isInstanceOf[Module], changed from MultiIOModule. Note that in the backport, we have to check MultiIOModule, not Module. I saw this as a bug, so just noting it here so we remember.